### PR TITLE
Llvm17 fix compiler-rt, add liblldb17

### PR DESCRIPTION
**Pull request by Calandracas606**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-musl
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64 (cross)
  - armv6l-musl (cross)

[ci skip]

### CODE REVIEW

File: srcpkgs/llvm17/template [hunk 1]
```diff
@@ -58,8 +59,9 @@ if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
 fi
 
 subpackages="libllvm17 llvm17-doc llvm17-devel"
-_enabled_projects=
-_enabled_runtimes=
+
+# use $(:) to silence xlint
+_enabled_projects=$(:)
```
Comment (classabbyamp) [inline srcpkgs/llvm17/template:64]: https://github.com/leahneukirchen/xtools/pull/323

Comment (leahneukirchen) [reply srcpkgs/llvm17/template:64]: or initialize it with `-DLLVM_ENABLE_PROJECTS=`, but i don't care strongly.

File: srcpkgs/llvm17/template [hunk 2]
```diff
@@ -576,17 +585,26 @@ libomp-devel_package() {
...
+			vmove usr/bin/llvm-omp-device-info
+		fi
+		if [ -f "${DESTDIR}/usr/bin/llvm-omp-kernel-replay" ]; then
+			vmove usr/bin/llvm-omp-kernel-replay
+		fi
+		if [ -f "${DESTDIR}/usr/lib/libarcher.so" ]; then
+			vmove "usr/lib/libarcher*.so"
+		fi
+		if [ -f "${DESTDIR}/usr/lib/libarcher_static.a" ]; then
+			vmove "usr/lib/libarcher*.a"
```
Comment (Johnnynator) [inline srcpkgs/llvm17/template:598]: This file was already moved 3 lines earlier. You should match the `vmove` with your condition

Comment (Calandracas606) [reply srcpkgs/llvm17/template:598]: the shared `.so` was moved 3 lines earlier, this vmoves the static `.a`

Comment (Johnnynator) [reply srcpkgs/llvm17/template:598]: True.

**Pull request closed without merging**